wallet/src/repository/wallet/mod.rs

```diff
@@ -588,47 +627,96 @@ where
+                db_utxo_set.insert(pointer.clone(), key_balance.clone());
+            }
             state.transaction_next_id = txn_next_id;
             state.balance = new_balance;
 
             // DB write batch
             let mut batch = self.db.batch();
-            let txn_hash = txn.hash();
+            let txn_hash = txn.transaction.hash();
 
-            // Transaction data
-            batch.put(
-                &keys::transaction_value(account, txn_id),
-                transaction_mutation.txn_balance,
-            )?;
-            batch.put(
-                keys::transaction_type(account, txn_id),
-                model::TransactionType::try_from(txn).map_err(|error| {
-                    repository::error::Error::UnsupportedTransactionType(error.0)
-                })?,
+            // Transaction indexes
+            batch.put(keys::transactions_index(txn_hash.as_ref()), txn_id)?;
+            batch.put(keys::transaction_hash(account, txn_id), txn_hash.as_ref())?;
+
+            // Miner fee
+            let miner_fee: u64 = match &txn.metadata {
+                Some(model::TransactionMetadata::InputValues(input_values)) => {
+                    let total_input_amount = input_values.iter().fold(0, |acc, x| acc + x.value);
+                    let total_output_amount = outputs.iter().fold(0, |acc, x| acc + x.value);
+
+                    total_input_amount
+                        .checked_sub(total_output_amount)
+                        .unwrap_or_else(|| {
+                            log::warn!("Miner fee below 0 in a transaction of type value transfer or data request: {:?}", txn.transaction);
+
+                            0
+                        })
+                }
+                _ => 0,
+            };
+
+            // Transaction Movement
+            let balance_movement = build_balance_movement(
+                &txn,
+                miner_fee,
+                account_mutation.kind,
+                account_mutation.amount,
+                &block_info,
+                convert_block_epoch_to_timestamp(state.epoch_constants, block_info.epoch),
             )?;
             batch.put(
-                keys::transaction_timestamp(account, txn_id),
-                convert_block_epoch_to_timestamp(state.epoch_constants, block_info.epoch),
+                keys::transaction_movement(account, txn_id),
+                balance_movement,
             )?;
-            batch.put(keys::transaction_block(account, txn_id), block_info)?;
 
-            // Transaction indexes
-            batch.put(keys::transactions_index(txn_hash.as_ref()), txn_id)?;
-            batch.put(keys::transaction_hash(account, txn_id), txn_hash.as_ref())?;
+            // Try to update data request tally report if it was previously indexed
+            if let types::Transaction::Tally(tally) = &txn.transaction {
+                log::debug!("Updating data request transaction with report from tally transaction");
```

Comment:
Given that this is `debug` level logging, let's take the opportunity to state which specific request and tally transactions were involved in this.